src/emoji/data.js

```diff
@@ -47,53 +49,40 @@ export const emojiTypeFromReactionType = (reactionType: ReactionType): EmojiType
  */
 export const getFilteredEmojis = (
   query: string,
-  activeImageEmojiByName: $ReadOnly<{| [string]: ImageEmojiType |}>,
-): $ReadOnlyArray<{| emoji_type: EmojiType, name: string, code: string |}> => {
-  // We start by making a map from matching emoji names to a number
-  // representing how good a match it is: 0 for a prefix match, 1 for a
-  // match anywhere else in the string.
-
-  const matchingUnicodeEmoji = objectEntries(unicodeCodeByName)
-    .map(([name, code]) => {
-      // This logic does not do any special handling for things like
-      // skin-tone modifiers or gender modifiers, since Zulip does not
-      // currently support those: https://github.com/zulip/zulip/issues/992.
-      // Once support is added for that, we may want to come back here and
-      // modify this logic, if for instance, there is a default skin-tone
-      // setting in the webapp that we want to also surface here. (or
-      // perhaps it will be best to leave it as is - that's a product
-      // decision that's yet to be made.) For the time being, it seems
-      // better to not show the user anything if they've searched for an
-      // emoji with a modifier than it is to show them the non-modified
-      // emoji, hence the very simple matching.
-      const matchesEmojiLiteral = parseUnicodeEmojiCode(code) === query;
-      const matchesEmojiName = Math.min(1, name.indexOf(query));
```

Comment:
emoji [nfc]: Use shared emoji matcher.
    
    Use typeahead.get_emoji_matcher from @zulip/shared to find matching
    emoji.

Is this really NFC -- does the shared matcher have the same behavior as this code? How did you investigate to determine that?

The answer should end up in the commit message, even if we also discuss it here on the PR or in a chat thread. That way it's there for future readers. See style guide:
https://github.com/zulip/zulip-mobile/blob/main/docs/style.md#commit-message

Reply:
Could you just explain what you want to do here? Would removing [nfc] be sufficient? At the same time, there are a handful of tests that test this exact functionality, yet none of them have changed. So who's to say it wouldn't be more confusing for future readers as to why it's NOT nfc?

Reply:
I removed the NFC label. I'll stare at this one a little more tomorrow, but my fear is that I'll just be tempted to re-write the tests for the shared code that already exist in https://github.com/zulip/zulip/blob/main/frontend_tests/node_tests/typeahead.js, in which there are already overlapping tests for functionality in mobile's tests https://github.com/zulip/zulip-mobile/blob/main/src/emoji/__tests__/data-test.js

Reply:
Yeah, reading the shared code's tests can be a useful source. Remember that they aren't necessarily real thorough, though -- at a glance, those look probably less thorough than the corresponding tests we have in mobile, and as we've seen, those don't necessarily catch all interesting changes either.

So I'd highly recommend reading the implementation itself to try to get a strong handle on its behavior.

One lens that may be useful when doing that is to think about tests for it -- i.e., to try to think up what test cases you'd want to write to exercise all the different things it's doing.